File: VinIDPayDemo/VinIDPayDemo/ViewController.m
//
//  ViewController.m
//  VinIDPayDemo
//
//  Created by Huong Do on 10/25/19.
//  Copyright © 2019 VinID. All rights reserved.
//

#import "ViewController.h"
#import <VinIDPaySDK/VinIDPaySDK.h>

@interface ViewController ()

@property (weak, nonatomic) IBOutlet UITextField *orderIdTextField;
@property (weak, nonatomic) IBOutlet UITextField *signatureTextField;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (IBAction)proceedPayment:(id)sender {
    [self.view endEditing:YES];
    [[VinIDPay sharedInstance] payWithOrderId:self.orderIdTextField.text signature:self.signatureTextField.text extraData:nil completionHandler:^(NSString *orderId, VinIDPayStatus status) {
        
        NSString *message = @"";
        
        switch (status) {
            case VinIDPayStatusAborted:
                message = @"payment aborted";
                break;
            
            case VinIDPayStatusFailure:
                message = @"payment failed";
                break;
                
            case VinIDPayStatusSuccess:
                message = @"payment successful";
                break;
        }
        
        UIAlertController *controller = [UIAlertController alertControllerWithTitle:nil message:message preferredStyle:UIAlertControllerStyleAlert];
        UIAlertAction *cancelAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleCancel handler:nil];
        [controller addAction:cancelAction];
        [self presentViewController:controller animated:YES completion:nil];
    }];
}

@end

File: SwiftDemo/SwiftDemo/ViewController.swift
//
//  ViewController.swift
//  SwiftDemo
//
//  Created by Huong Do on 10/27/19.
//  Copyright © 2019 VinID. All rights reserved.
//

import UIKit
import VinIDPaySDK

class ViewController: UIViewController {
    
    @IBOutlet weak var orderTextField: UITextField!
    @IBOutlet weak var signatureTextField: UITextField!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
    }
    
    @IBAction func checkAppInstall(_ sender: Any) {
        let hasInstalled = VinIDPay.sharedInstance.hasVinIDAppInstalled
        let message = hasInstalled ? "App installed, ready to pay" : "App not installed"
        
        let controller = UIAlertController(title: nil, message: message, preferredStyle: .alert)
        let action = UIAlertAction(title: "OK", style: .cancel, handler: nil)
        controller.addAction(action)
        present(controller, animated: true, completion: nil)
    }
    
    @IBAction func proceedPayment(_ sender: Any) {
        view.endEditing(true)
        guard let id = orderTextField.text,
            let signature = signatureTextField.text else { return }
        
        VinIDPay.sharedInstance.pay(withOrderId: id, signature: signature) { [weak self] (id, status) in
            var message = ""
            
            switch status {
            case .success:
                message = "payment successful!"
            case .aborted:
                message = "user aborted payment"
            case .failure:
                message = "payment failed"
            }
            
            let controller = UIAlertController(title: nil, message: message, preferredStyle: .alert)
            let action = UIAlertAction(title: "OK", style: .cancel, handler: nil)
            controller.addAction(action)
            self?.present(controller, animated: true, completion: nil)
        }
    }
    
}

